src/workspaceConfig.ts

```diff
@@ -45,16 +44,27 @@ export function updateIdfComponentsTree(projectDescriptionPath: string) {
     idfDataProvider.refresh(projectDescriptionPath);
 }
 
-export function updateProjectName(projectDescriptionPath: string) {
-    if (!utils.fileExists(projectDescriptionPath)) {
-        return;
-    }
-    fs.readFile(projectDescriptionPath, (err, data) => {
-        if (err) {
-            Logger.errorNotify(err.message, err);
-            return;
+export function getProjectName(workspace: string): Promise<string> {
+    const cmakeListFile = path.join(workspace, "CMakeLists.txt");
+    return new Promise((resolve, reject) => {
+        try {
+            if (!utils.fileExists(cmakeListFile)) {
+                reject(`${cmakeListFile} doesn't exist.`);
+            }
+            fs.readFile(cmakeListFile, (err, data) => {
+                if (err) {
+                    Logger.errorNotify(err.message, err);
+                    reject(err);
+                }
+                const content = data.toString();
+                const projectMatches = content.match(/(?:project\()(.*?)(?:\))/);
```

Comment:
Ouch, this seems to be very fragile... if the argument to the project function is a variable reference, then this won't work. Any chance we can get the project name from the project metadata file instead (build/project_description.json, "project_name" key). The project metadata file becomes available after CMake configuration phase runs.

Reply:
Also, If we use `build/project_description.json` then we can do `vscode.workspace.createFileSystemWatcher` for watching the `CMakeLists.txt` and run `CMake configuration phase` on change and update the project name, if we want to add kind of instant update

Reply:
Yes, i think this is what VS Code CMake extension does — it watches CMakeLists.txt files and re-runs CMake when the file changes.

If you do end up parsing the project description, you can get more things from it. For instance, the list of project components, the path of the sdkconfig file, toolchain prefix, monitor baud rate, and so on.

Reply:
Thanks for the feedback, will add them soon.